### PR TITLE
fix: always finish progress in autofixing

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

_Provide description of this PR and changes, if linked Jira ticket doesn't cover it in full._

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

🚨After having merged, please update the CLI go.mod to pull in latest language server.

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: infrastructure/code/issue_enhancer.go [hunk 1]
```diff
@@ -187,15 +189,18 @@ func (b *IssueEnhancer) autofixFunc(ctx context.Context, issue snyk.Issue,
...
 			select {
 			case <-timeoutTimer.C:
 				log.Error().Str("method", "GetAutofixSuggestions").Str("requestId", b.requestId).Msg("timeout requesting autofix")
 				b.notifier.SendShowMessage(sglsp.MTError, "Something went wrong. Please try again. Request ID: "+b.requestId)
 				return nil
 			case <-pollingTicker.C:
+				p.ReportWithMessage(int(math.Max(tries, 99)), "Polling for fix...")
 				fix, complete := pollFunc()
 				if !complete {
+					tries++
```
Comment (cat2608) [inline infrastructure/code/issue_enhancer.go:203]: Is this logic to set a ceiling to the max number or retries?

Comment (bastiandoetsch) [reply infrastructure/code/issue_enhancer.go:203]: Nope, it's just to not report more than a 100% (timeout is currently set to 120, and didn't want to implement a complicated logic with rounding to integers afterwards).

File: infrastructure/code/issue_enhancer.go [hunk 2]
```diff
@@ -187,15 +189,18 @@ func (b *IssueEnhancer) autofixFunc(ctx context.Context, issue snyk.Issue,
...
 			select {
 			case <-timeoutTimer.C:
 				log.Error().Str("method", "GetAutofixSuggestions").Str("requestId", b.requestId).Msg("timeout requesting autofix")
 				b.notifier.SendShowMessage(sglsp.MTError, "Something went wrong. Please try again. Request ID: "+b.requestId)
 				return nil
 			case <-pollingTicker.C:
+				p.ReportWithMessage(int(math.Min(tries, 99)), "Polling for fix...")
 				fix, complete := pollFunc()
 				if !complete {
+					tries++
```
Comment (cat2608) [inline infrastructure/code/issue_enhancer.go:203]: I think this is going it be showing 99 until tries passes 99, right? We are not aiming to show a progress I understand

**Review by cat2608 — APPROVED** [review]

File: infrastructure/code/issue_enhancer.go (hunk 2 above)
Comment (bastiandoetsch) [reply infrastructure/code/issue_enhancer.go:203]: Now, it's gonna show a progress up to the 99s limit - with Max it was only showing 99...

**Pull request merged**